lib/api/CDataFrameAnalysisInstrumentation.cc

```diff
@@ -89,11 +94,15 @@ void CDataFrameAnalysisInstrumentation::writeProgress(std::uint32_t step) {
     }
 }
 
-void CDataFrameAnalysisInstrumentation::writeMemory(std::uint32_t step) {
+void CDataFrameAnalysisInstrumentation::writeMemory(std::int64_t timestamp) {
     if (m_Writer != nullptr) {
         m_Writer->StartObject();
-        m_Writer->Key(STEP_TAG);
-        m_Writer->Uint(step);
+        m_Writer->Key(TYPE_TAG);
```

Comment:
This doesn't lead to the expected format. Java expects named top level objects, so in this case we should expect:

```
"analysis_memory_usage": {
  "job_id": "foo",
  "timestamp: 123566778,
  "peak_usage_bytes": 42
}
```

Reply:
I updated the code as we discussed.